File: 프로그래머스/lv1/12977. 소수 만들기/소수 만들기.py
from itertools import combinations
def solution(nums):
    
    num = list(combinations(nums, 3))
    cnt = 0
    for i in num:
        tmp = 0
        for j in range(2, sum(i) + 1):
            if sum(i) % j == 0:
                tmp += 1
            
        if tmp == 1:
            cnt += 1
    
    return cnt
    
    

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.java
import java.util.*;
class Solution {

    int answer = 0;

    public int solution(int[] nums) {
        comb(nums, 0, 0, 0);
        return answer;
    }

    void comb(int[] nums, int start, int cnt, int sum) {

        if (cnt == 3) {
            if (isPrime(sum)) {
                answer++;
            }
            return;
        }

        for (int i = start; i < nums.length; i++) {
            comb(nums, i + 1, cnt + 1, sum + nums[i]);
        }
    }

    boolean isPrime(int n) {
        if (n < 2) return false;

        for (int i = 2; i * i <= n; i++) {
            if (n % i == 0) {
                return false;
            }
        }

        return true;
    }
}